catalyst-toolbox/src/stats/voters.rs

```diff
@@ -36,21 +32,87 @@ pub fn calculate_wallet_distribution<S: Into<String>>(
                 chain_addr::Address(discrimination, Kind::Account(public)),
             )
         })
-        .collect();
+        .collect()
+}
+
+fn vote_counts_as_addresses(
+    votes_count: VoteCount,
+    genesis: &Block0Configuration,
+) -> Vec<(InitialUTxO, u32)> {
+    genesis
+        .initial
+        .iter()
+        .filter_map(|initials| {
+            if let Initial::Fund(funds) = initials {
+                for utxo in funds {
+                    if let Some((_, votes_count)) = votes_count.iter().find(|(address, _)| {
+                        account_hex_to_address(
+                            address.to_string(),
+                            genesis.blockchain_configuration.discrimination,
+                        )
+                        .unwrap()
+                            == utxo.address
+                    }) {
+                        return Some((utxo.clone(), *votes_count as u32));
```

Comment:
the iteration through `votes_count` is still a bit inefficient, I think we can go from the address to the hex key and access `votes_count` as a map directly  e.g.
```
for utxo in funds {
    if let Some(vote_count) = votes_count.get(addr_to_hex(utxo.address)) {
        return Some(utxo.clone(), vote_count)
    }
}
```